plugins/autoslab.cpp

```diff
@@ -0,0 +1,267 @@
+/* Simple plugin to check for ghosts and automatically queue jobs to engrave slabs for them.
+ *
+ * Enhancement idea: Queue up a ConstructSlab job, then link the engrave slab job to it. Avoids need to have slabs in stockpiles
+ *                   Would require argument parsing, specifying materials
+ * Enhancement idea: Automatically place the slab. This seems like a tricky problem but maybe solveable with named zones?
+ *                   Might be made obsolete by people just using buildingplan to pre-place plans for slab?
+ * Enhancement idea: Optionally enable autoengraving for pets.
+ * Enhancement idea: Try to get ahead of ghosts by autoengraving for dead dwarves with no remains.
+ */
+
+#include "Core.h"
+#include "Debug.h"
+#include "PluginManager.h"
+
+#include "modules/Persistence.h"
+#include "modules/Translation.h"
+#include "modules/World.h"
+
+#include "df/historical_figure.h"
+#include "df/item.h"
+#include "df/manager_order.h"
+#include "df/plotinfost.h"
+#include "df/unit.h"
+#include "df/world.h"
+
+using namespace DFHack;
+
+static command_result autoslab(color_ostream &out, std::vector<std::string> &parameters);
+
+DFHACK_PLUGIN("autoslab");
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+REQUIRE_GLOBAL(world);
+
+// logging levels can be dynamically controlled with the `debugfilter` command.
+namespace DFHack
+{
+    // for configuration-related logging
+    DBG_DECLARE(autoslab, status, DebugCategory::LINFO);
+    // for logging during the periodic scan
+    DBG_DECLARE(autoslab, cycle, DebugCategory::LINFO);
+}
+
+static const auto CONFIG_KEY = std::string(plugin_name) + "/config";
+static PersistentDataItem config;
+enum ConfigValues
+{
+    CONFIG_IS_ENABLED = 0,
+    CONFIG_CYCLE_TICKS = 1,
+};
+static int get_config_val(int index)
+{
+    if (!config.isValid())
+        return -1;
+    return config.ival(index);
+}
+static bool get_config_bool(int index)
+{
+    return get_config_val(index) == 1;
+}
+static void set_config_val(int index, int value)
+{
+    if (config.isValid())
+        config.ival(index) = value;
+}
+static void set_config_bool(int index, bool value)
+{
+    set_config_val(index, value ? 1 : 0);
+}
+
+static int32_t cycle_timestamp = 0; // world->frame_counter at last cycle
+
+static command_result do_command(color_ostream &out, std::vector<std::string> &parameters);
+static void do_cycle(color_ostream &out);
+
+DFhackCExport command_result plugin_init(color_ostream &out, std::vector<PluginCommand> &commands)
+{
+    DEBUG(status, out).print("initializing %s\n", plugin_name);
+
+    // provide a configuration interface for the plugin
+    commands.push_back(PluginCommand(
+        plugin_name,
+        "Automatically engrave slabs of ghostly citizens!",
+        do_command));
```

Comment:
since you have no configuration you can take this out. you've already marked the tool as `:no-command:` in the docs.